src/formats/gaussformat.cpp

```diff
@@ -711,10 +725,64 @@ namespace OpenBabel
                 if (!atom)
                   break;
                 atom->SetPartialCharge(atof(vs[2].c_str()));
-
+                Mulliken->AddPartialCharge(atof(vs[2].c_str()));
                 if (!ifs.getline(buffer,BUFF_SIZE)) break;
                 tokenize(vs,buffer);
+                                    
+              } 
+            if (Mulliken->NumPartialCharges() != mol.NumAtoms()) 
+            {
+                cerr << "The number of Mulliken partial charges is not equal to the number of atoms!" << endl;
+                exit (-1);
+            }
+            Mulliken->SetAttribute("Mulliken charges");
+            mol.SetData(Mulliken);    
+          }
+        else if(strstr(buffer,"Hirshfeld charges") != NULL &&
+                strstr(buffer,"CM5 charges") != NULL)
+          {
+            /*
+              Hirshfeld and CM5 charges are printed in the
+              same block in the Gaussian log file. 
+             */
+            hasPartialCharges = true;
+            chargeModel = "Hirshfeld";
+            if (mol.HasData("Hirshfeld charges"))
+              {
+                mol.DeleteData("Hirshfeld charges");
+              }
+            if (mol.HasData("CM5 charges"))
+              {
+                mol.DeleteData("CM5 charges");
               }
+            OBPcharge *Hirshfeld = new OpenBabel::OBPcharge();
+            OBPcharge *CM5       = new OpenBabel::OBPcharge();
+            ifs.getline(buffer,BUFF_SIZE);	// column headings
+            ifs.getline(buffer,BUFF_SIZE);
+            tokenize(vs,buffer);
+            while (vs.size() >= 8 &&
+                   strstr(buffer,"Tot ") == NULL)
+              {
+                atom = mol.GetAtom(atoi(vs[0].c_str()));
+                if (!atom)
+                  break;
+                atom->SetPartialCharge(atof(vs[2].c_str()));
+                Hirshfeld->AddPartialCharge(atof(vs[2].c_str()));
+                CM5->AddPartialCharge(atof(vs[7].c_str()));
+                if (!ifs.getline(buffer,BUFF_SIZE)) break;
+                tokenize(vs,buffer);
+                                    
+              }
+            if (CM5->NumPartialCharges() != mol.NumAtoms() || 
+                Hirshfeld->NumPartialCharges() != mol.NumAtoms()) 
+            {
+                cerr << "The number of Hirshfeld or CM5 partial charges is not equal to the number of atoms!" << endl;
+                exit (-1);
+            }
+            Hirshfeld->SetAttribute("Hirshfeld charges");
+            CM5->SetAttribute("CM5 charges");
```

Comment:
These should all have:
CM5->SetOrigin(fileformatInput);

(as in, they're not perceived Gasteiger or other charges)